drivers/sensor/s11059/s11059.c

```diff
@@ -0,0 +1,313 @@
+		return -ENOTSUP;
+	}
+
+	rc = s11059_start_measurement(dev);
+	if (rc < 0) {
+		LOG_ERR("%s, Failed to start measurement.", dev->name);
+		return rc;
+	}
+
+	do {
+		rc = i2c_reg_read_byte_dt(&cfg->bus, S11059_REG_ADDR_CONTROL, &control);
+		if (rc < 0) {
+			LOG_ERR("%s, Failed to read control.", dev->name);
+			return rc;
+		}
+		k_usleep(S11059_WAIT_PER_LOOP);
+
+	} while (!(control & S11059_BIT_MASK_CONTROL_STANDBY_MONITOR));
+
+	rc = s11059_samples_read(dev, S11059_REG_ADDR_DATA, values, sizeof(values));
+	if (rc < 0) {
+		LOG_ERR("%s, Failed to get sample.", dev->name);
+		return rc;
+	}
+
+	for (size_t i = 0; i < NUM_OF_COLOR_CHANNELS; i++) {
+		drv_data->samples[i] = values[i];
+	}
+
+	return 0;
+}
+
+static int s11059_channel_get(const struct device *dev, enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	const struct s11059_dev_config *cfg = dev->config;
+	struct s11059_data *drv_data = dev->data;
+	const uint8_t index = s11059_convert_channel_to_index(chan);
+	const uint16_t factor = convert_factors[cfg->gain][index];
+	uint32_t meas_value;
+
+	meas_value = drv_data->samples[index] * S11059_CARRY_UP / factor;
+	val->val1 = meas_value / (S11059_CARRY_UP / 10);
+	val->val2 = meas_value % (S11059_CARRY_UP / 10);
+
+	return 0;
+}
+
+static int s11059_init(const struct device *dev)
+{
+	const struct s11059_dev_config *cfg = dev->config;
+	uint8_t control = S11059_INITIAL_CONTROL;
+	uint16_t manual_time;
+	uint8_t timing_mode;
+	int rc;
+
+	/* device set */
+	if (!device_is_ready(cfg->bus.bus)) {
+		LOG_ERR("%s, device is not ready.", dev->name);
+		return -ENODEV;
+	}
+
+	rc = s11059_integ_time_calculate(dev, &manual_time, &timing_mode);
+	if (rc < 0) {
+		LOG_ERR("%s, Failed to calculate manual timing.", dev->name);
+		return rc;
+	}
+
+	rc = s11059_manual_timing_write(dev, manual_time);
+	if (rc < 0) {
+		LOG_ERR("%s, Failed to set manual timing.", dev->name);
+		return rc;
+	}
+
+	/* set integration time mode and gain*/
+	control |= timing_mode & S11059_BIT_MASK_INTEGRATION_TIME;
+	WRITE_BIT(control, S11059_CONTROL_GAIN, cfg->gain);
+	rc = s11059_control_write(dev, control);
+	if (rc < 0) {
+		LOG_ERR("%s, Failed to set gain and integration time.", dev->name);
+		return rc;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api s11059_driver_api = {
+	.sample_fetch = s11059_sample_fetch,
+	.channel_get = s11059_channel_get,
+};
+
+#define S11059_INST(inst)                                                                          \
+	static struct s11059_data s11059_data_##inst;                                              \
+	static const struct s11059_dev_config s11059_config_##inst = {                             \
+		.bus = I2C_DT_SPEC_INST_GET(inst),                                                 \
+		.gain = DT_INST_PROP_OR(inst, gain, 0),                                            \
+		.integration_time = DT_INST_PROP_OR(inst, integration_time, 546000)};              \
```

Comment:
How did you choose this default value?

Reply:
I set these default values by referring to the initial settings in the sensor's datasheet.
Please refer to page 3 of the datasheet for that.

Datasheet: https://datasheetspdf.com/pdf/1323325/Hamamatsu/S11059-02DT/1